jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/Stream.java

```diff
@@ -0,0 +1,376 @@
+         *             Content.Chunk chunk = stream.read();
+         *             if (chunk == null)
+         *             {
+         *                 // No data available now, demand to be called back.
+         *                 stream.demand();
+         *                 return;
+         *             }
+         *
+         *             // Process the content chunk.
+         *             process(chunk);
+         *
+         *             // Notify that the content has been consumed.
+         *             chunk.release();
+         *
+         *             if (chunk.isLast())
+         *             {
+         *                 // All data has been processed.
+         *                 return;
+         *             }
+         *         }
+         *     }
+         * }
+         * }</pre>
+         *
+         * @param stream the stream
+         * @see Stream#demand()
+         */
+        default void onDataAvailable(Stream stream)
+        {
+            while (true)
+            {
+                Content.Chunk chunk = stream.read();
+                if (chunk == null)
+                {
+                    stream.demand();
+                    return;
+                }
+                chunk.release();
+                if (chunk.isLast())
+                    return;
+            }
+        }
+
+        /**
+         * <p>Invoked when a STREAM_DATA_BLOCKED frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onDataBlocked(Stream stream, StreamDataBlockedFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a MAX_STREAM_DATA frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onMaxData(Stream stream, StreamMaxDataFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a STOP_SENDING frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onStopSending(Stream stream, StopSendingFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a RESET_STREAM frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onReset(Stream stream, ResetFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when the stream has been {@link Stream#isClosed() closed}.</p>
+         *
+         * @param stream the stream
+         */
+        default void onClose(Stream stream)
```

Comment:
is this called before or after an EOF chunk is read?

Reply:
Updated javadocs, clarifying when this method is called.